udata/core/dataset/api.py

```diff
@@ -550,3 +551,18 @@ def get(self):
         '''List all resource types'''
         return [{'id': id, 'label': label}
                 for id, label in RESOURCE_TYPES.items()]
+
+
+@ns.route('/schemas/', endpoint='schemas')
+class SchemasAPI(API):
+    @api.doc('schemas')
+    @api.marshal_list_with(schema_fields)
+    def get(self):
+        '''List all available schemas'''
+        try:
+            # This method call is cached as it makes HTTP requests
+            return ResourceSchema.objects()
+        except LookupError:
+            abort(503, description='No schemas in cache and endpoint unavailable')
+        except ValueError:
```

Comment:
It was difficult to pick appropriate exceptions to raise, I'm open to suggestions

Reply:
I wouldn't try to use base exceptions that are not made for that.

Either:
- use custom exceptions
- use the return value of the function (`[]` vs `None` for example) if it's enough

Reply:
I added custom exceptions 🔧